src/status_im/ui/screens/events.cljs

```diff
@@ -401,8 +407,14 @@
 
 (handlers/register-handler-fx
   :app-state-change
-  (fn [_ [_ state]]
-    (status/app-state-change state)))
+  (fn [{{:keys [network-status mailserver-status]} :db :as cofx} [_ state]]
+    (let [app-coming-from-background? (= state "active")
+          should-recover? (and app-coming-from-background?
+                               (= network-status :online)
```

Comment:
so here we need to be careful in case when 
1) app enters background
2) after 2-3 minutes we lost connection
3) app enters foreground

The problem is that `app-state:active` event may appear before `connection:offline` and we will request messages while being offline. At least this case needs to be tested, maybe it's covered somehow with `mailserver-status` check.

Reply:
>after 2-3 minutes we lost connection

@rasom i was not able to reproduce this during manual testing. Even if i disable internet in background, enable it back and go foreground, mailserver connection gets recovered after several retries.